app/_src/gateway-operator/production/workloads-autoscaling/prometheus.md

```diff
@@ -0,0 +1,225 @@
+   ```
+
+1. Install `prometheus-adapter` via helm chart
+
+   ```bash
+   helm upgrade --install --create-namespace -n prometheus --values values.yaml prometheus-adapter prometheus-community/prometheus-adapter
+   ```
+
+## Verify metrics are exposed in Kubernetes
+
+When all is configured you should be able to see the metric you've configure in `prometheus-adapter` exposed via Kubernetes Custom Metrics API:
+
+```bash
+kubectl get --raw "/apis/custom.metrics.k8s.io/v1beta1/namespaces/default/services/echo/kong_upstream_latency_ms_30s_average" | jq
+```
+
+This should result in:
+
+```json
+{
+  "kind": "MetricValueList",
+  "apiVersion": "custom.metrics.k8s.io/v1beta1",
+  "metadata": {},
+  "items": [
+    {
+      "describedObject": {
+        "kind": "Service",
+        "namespace": "default",
+        "name": "echo",
+        "apiVersion": "/v1"
+      },
+      "metricName": "kong_upstream_latency_ms_30s_average",
+      "timestamp": "2024-03-06T13:11:12Z",
+      "value": "0",
+      "selector": null
+    }
+  ]
+}
+```
+
+## Use exposed metric in `HorizontalPodAutoscaler`
+
+When the metric configured in `prometheus-adapter` is available through Kubernetes' Custom Metrics API
+we can use it in `HorizontalPodAutoscaler` to autoscale our workload: specifically the `echo` `Service`.
+
+This can be done by using the following manifest:
+
+```yaml
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: echo
+  namespace: default
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: echo
+  minReplicas: 1
+  maxReplicas: 10
+  behavior:
+    scaleDown:
+      stabilizationWindowSeconds: 1
+      policies:
+      - type: Percent
+        value: 100
+        periodSeconds: 10
+    scaleUp:
+      stabilizationWindowSeconds: 1
+      policies:
+      - type: Percent
+        value: 100
+        periodSeconds: 2
+      - type: Pods
+        value: 4
+        periodSeconds: 2
+      selectPolicy: Max
+  metrics:
+  - type: Object
+    object:
+      metric:
+        name: "kong_upstream_latency_ms_30s_average"
+      describedObject:
+        apiVersion: v1
+        kind: Service
+        name: echo
+
+      target:
+        type: Value
+        value: "40"
+```
+
+This configuration will scale the underlying `echo` `Deployment` between 1 and 10 replicas, trying to keep the average latency
+across last 30s at 40ms.
+
+You can watch those events using the following `kubectl` command:
+
+```bash
+kubectl get events --field-selector involvedObject.name=echo --field-selector involvedObject.kind=HorizontalPodAutoscaler -w
+```
+
+Assuming that you have access to the deployed `Gateway` addres you can try enforcing the scaling by issuing requests like so:
+
+```bash
+while curl -k http://$(kubectl get gateway kong -o custom-columns='name:.status.addresses[0].value' --no-headers)/echo/shell\?cmd=sleep%200.1 ; sleep 1; do ; done
```

Comment:
```suggestion
while curl -k "http://$(kubectl get gateway kong -o custom-columns='name:.status.addresses[0].value' --no-headers -n default)/echo/shell\?cmd=sleep%200.1"; echo ; sleep 1; do true; done
```

- quoting (copy-pasting didn't work on my config of zsh with autoscaping on paste, now it works bot way)
- as far I saw all yamls commands previously do not rely on current namespace thus added `-n default` to make it follow the pattern
- additional `echo` to have
```
NOW: 2024-03-06 15:58:50.238007532 +0000 UTC m=+2476.353484040
NOW: 2024-03-06 15:58:51.415274089 +0000 UTC m=+2477.530750556
NOW: 2024-03-06 15:58:52.810841024 +0000 UTC m=+2478.926317615
```

instead of

```
NOW: 2024-03-06 15:56:41.704484172 +0000 UTC m=+2347.819960639NOW: 2024-03-06 15:56:42.816962693 +0000 UTC m=+2348.932439118NOW: 2024-03-06 15:56:43.911404608 +0000 UTC m=+2350.026881075NOW: 2024-03-06 15:56:45.013871361 +0000 UTC m=+2351.129347827
```